### PR TITLE
Add tests for various classes in the broker maven sub-module.

### PR DESCRIPTION
**Pull request by johanneskloos**
I've analysed your codebase and noticed that some classes in the broker sub-module are not fully tested.
I've written some tests for the methods in this class with the help of [Diffblue Cover](https://www.diffblue.com/opensource).
Hopefully, these tests will help you detect any regressions caused by future code changes. If you would find it useful to have additional tests written for this repository, I would be more than happy to look at other classes that you consider important in a subsequent PR.

### CODE REVIEW

**Review by JohnLBergqvist — CHANGES_REQUESTED** [review]
Fix your POM file first. I don't see why it's showing the entire file as changed.

File: broker/src/test/java/com/xiaoleilu/loServer/handler/HttpResponseHelperTest.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package com.xiaoleilu.loServer.handler;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class HttpResponseHelperTest {
+    @Test
+    public void getMultipartUploadResponseBody() throws Exception {
+        final String remoteFileName = "foo.txt";
+        final String fileUrl = "http://www.google.com/";
```
Comment (JohnLBergqvist) [inline broker/src/test/java/com/xiaoleilu/loServer/handler/HttpResponseHelperTest.java]: Never use third party URLs - Use something like `foo.bar`

**Review by JohnLBergqvist — CHANGES_REQUESTED** [review]

File: broker/src/test/java/win/liyufan/im/UtilityTest.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class UtilityTest {
+    @Test
+    public void formatJsonInput() {
+        Assert.assertEquals("", Utility.formatJson(null));
+        Assert.assertEquals("1", Utility.formatJson("1"));
+        // Hand-written, as inspired by generated tests
```
Comment (JohnLBergqvist) [inline broker/src/test/java/win/liyufan/im/UtilityTest.java]: Remove this comment.

File: broker/src/test/java/com/xiaoleilu/loServer/handler/HttpResponseHelperTest.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        final String actual =
+            HttpResponseHelper.getMultipartUploadResponseBody(remoteFileName, fileUrl);
+
+        Assert.assertEquals("{\"rc_url\":{\"path\":\"http://www.google.com/\",\"type\":0}}", actual);
+    }
+
+    @Test
+    public void getFileExt() {
+        Assert.assertEquals("野火", HttpResponseHelper.getFileExt("...野火"));
+        Assert.assertEquals("", HttpResponseHelper.getFileExt("look_no_dots"));
```
Comment (JohnLBergqvist) [inline broker/src/test/java/com/xiaoleilu/loServer/handler/HttpResponseHelperTest.java]: use `foo` here.

File: broker/src/test/java/com/xiaoleilu/loServer/handler/HttpResponseHelperTest.java (hunk 1 above)
Comment (xbauch) [reply broker/src/test/java/com/xiaoleilu/loServer/handler/HttpResponseHelperTest.java]: Changed to `file:somefile`.

**Comment by xbauch** [comment]
Turned out we don't need to modify the `pom.xml` file at all.

**Review by JohnLBergqvist — CHANGES_REQUESTED** [review]

File: broker/src/test/java/com/xiaoleilu/loServer/handler/HttpResponseHelperTest.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class HttpResponseHelperTest {
+    @Test
+    public void getMultipartUploadResponseBody() throws Exception {
+        final String remoteFileName = "foo.txt";
+        final String fileUrl = "file:somefile";
+
```
Comment (petersear) [inline broker/src/test/java/com/xiaoleilu/loServer/handler/HttpResponseHelperTest.java]: remove blank line

File: broker/src/test/java/win/liyufan/im/UtilityTest.java (hunk 2 above)
Comment (petersear) [inline broker/src/test/java/win/liyufan/im/UtilityTest.java]: Maybe remove this comment. We don't tend to use the word `generated`.
Although the tests were initially created by Diffblue Cover they have been subsequently tailored to the repo so I don't think it's worth highlighting a difference between the tests' origins.

**Review by petersear — CHANGES_REQUESTED** [review]
Add a blank line and "These tests were written using Diffblue Cover." to the commit message.

File: broker/src/test/java/win/liyufan/im/UtilityTest.java (hunk 2 above)
Comment (xbauch) [reply broker/src/test/java/win/liyufan/im/UtilityTest.java]: Removed.

File: broker/src/test/java/com/xiaoleilu/loServer/handler/HttpResponseHelperTest.java (hunk 3 above)
Comment (xbauch) [reply broker/src/test/java/com/xiaoleilu/loServer/handler/HttpResponseHelperTest.java]: Done.

File: broker/src/test/java/com/xiaoleilu/loServer/handler/HttpResponseHelperTest.java (hunk 4 above)
Comment (xbauch) [reply broker/src/test/java/com/xiaoleilu/loServer/handler/HttpResponseHelperTest.java]: Removed.

File: broker/src/test/java/win/liyufan/im/UtilityTest.java (hunk 2 above)
Comment (xbauch) [reply broker/src/test/java/win/liyufan/im/UtilityTest.java]: Ok. Removed.

**Comment by xbauch** [comment]
> Add a blank line and "These tests were written using Diffblue Cover." to the commit message.

Done.

**Review by petersear — APPROVED** [review]

**Review by JohnLBergqvist — APPROVED** [review]

**Comment by johanneskloos** [comment]
Submitted externally, closing.

**Pull request closed without merging**